test/grpc/codec/json_test.exs

```diff
@@ -0,0 +1,99 @@
+defmodule GRPC.Codec.JSONTest do
+  use ExUnit.Case, async: true
+
+  alias GRPC.Codec.JSON
+
+  describe "name/0" do
+    test "returns the codec name" do
+      assert JSON.name() == "json"
+    end
+  end
+
+  describe "encode/1" do
+    test "encodes a protobuf struct to JSON" do
+      request = %Helloworld.HelloRequest{name: "Alice"}
+      encoded = JSON.encode(request)
+
+      assert is_binary(encoded)
+      decoded_map = Jason.decode!(encoded)
+      assert decoded_map["name"] == "Alice"
+    end
+
+    test "includes unpopulated fields with default values" do
+      request = %Helloworld.HelloRequest{name: "Bob", duration: nil}
+      encoded = JSON.encode(request)
+
+      decoded_map = Jason.decode!(encoded)
+      assert Map.has_key?(decoded_map, "duration")
+    end
+
+    test "encodes empty struct with all default fields" do
```

Comment:
Why encode all fields?

Reply:
I didn't understand which test you're referring to.